### DISCUSSION TITLE
Switching between Line-in and Mic on Audio Chips ES8388

### DISCUSSION THREAD

**Discussion by mkoenings** [pschatzmann/arduino-audiokit#40] [Ideas]
Hello,
I have a LyraT compatible board from Olimex  (works as Type: 1)
https://www.olimex.com/Products/IoT/ESP32/ESP32-ADF/
The guys did a great job but somehow "forgot" to wire out a second plug for Aux-in (Line-In)
Since I want to use the board for a Guitar effects project, the microphones are useless. I am considering to de-solder the mics, and switch the chip to process Line-In level signals (wiring to the LIN1, RIN1) , which seems to be possible. The datasheet (http://www.everest-semi.com/pdf/ES8388%20DS.pdf) shows different routings to the internal mixer, with a mic amp or without. 
But how can this be achieved ? 
I put this into "Ideas", maybe other projects on other boards can benefit from such an input switching feature.
Thanks and greetings from the Black Forest, 
Michael

**Comment by pschatzmann** [comment]
I am lacking of knowlege in electronics to give an answer. I was double checking how the other boards were dealing with this and I noticed that mics are wired differently then line in. See e.g. https://docs.ai-thinker.com/_media/esp32-audio-kit_v2.2_sch.pdf

So this will probably not work. 

Just in case I would also not recommend the AI Thinker AudioKit for this purpose because they messed up as well and you can't switch off the microphones when you want to get input from line in.

LyraT is working fine however in this respect!

**Reply by mkoenings** [reply]
thanks Phil. I also got Feedback from the Olimex people, they proposed to use the line2 input, but solder it straight to the pin of the chip, since it is not routed on the PCB :-) 
I ordered a LyraT, following your advice.
Thank you!
